dis_snek/models/discord/message.py

```diff
@@ -279,10 +282,20 @@ def _process_dict(cls, data: dict, client: "Snake") -> dict:
                 mention_ids.append(client.cache.place_user_data(user_data).id)
         data["mention_ids"] = mention_ids
 
-        if "mention_channels" in data:
+        extra = channel_mention.findall(data["content"])
+        if "mention_channels" in data or len(extra) > 0:
```

Comment:
The if logic needs to be separated, or not you will loop through data["mention_channels"] when it is None.

```
Traceback (most recent call last):
  File "c:\users\wben1\documents\github\dis_snek\dis_snek\api\events\processors\message_events.py", line 29, in _on_raw_message_create
    msg = self.cache.place_message_data(event.data)
  File "c:\users\wben1\documents\github\dis_snek\dis_snek\client\smart_cache.py", line 402, in place_message_data
    message = Message.from_dict(data, self._client)
  File "c:\users\wben1\documents\github\dis_snek\dis_snek\models\discord\base.py", line 24, in from_dict
    data = cls._process_dict(data, client)
  File "c:\users\wben1\documents\github\dis_snek\dis_snek\models\discord\message.py", line 288, in _process_dict
    for channel_data in data["mention_channels"]:
KeyError: 'mention_channels'
```

Reply:
Resolved both in latest commit